cmd/generate/generate.go

```diff
@@ -0,0 +1,276 @@
+	}
+
+	log.Debugf("source branch: %q, dest branch: %q\n", source, dest)
+
+	method, version := defineBumpStrategy(p.Bump, source, dest, p.MainBranchName, p.DevelopBranchName)
+
+	log.Debugf("method: %q, version: %q", method, version)
+
+	tag, err := getLatestTagOrDefault(p.Prefix)
+	if err != nil {
+		log.Errorf("failed getting latest tag: %s", err)
+	}
+
+	// Print current tag.
+	fmt.Printf("::set-output name=PREVIOUS_TAG::%s\n", p.Prefix+tag.String())
+
+	if tagSource != "git" {
+		tag = p.BaseVersion
+	}
+
+	if tagSource == "git" && version == "major" && method == "build" {
+		log.Debug("incrementing major")
+		if err := tag.IncrementMajor(); err != nil {
+			log.Errorf("failed to increment major version: %s", err)
+
+			os.Exit(exitcode.ErrDefault)
+		}
+	}
+
+	if tagSource == "git" && version == "minor" && method == "build" {
+		log.Debug("incrementing minor")
+		if err := tag.IncrementMinor(); err != nil {
+			log.Errorf("failed to increment minor version: %s", err)
+
+			os.Exit(exitcode.ErrDefault)
+		}
+	}
+
+	if tagSource == "git" && (version == "patch" && method == "build") || method == "hotfix" {
+		log.Debug("incrementing patch")
+		if err := tag.IncrementPatch(); err != nil {
+			log.Errorf("failed to increment patch version: %s", err)
+
+			os.Exit(exitcode.ErrDefault)
+		}
+	}
+
+	var finalTag string
+
+	if method == "build" {
+		buildNumber, _ := semver.NewPRVersion("0")
+
+		if len(tag.Pre) > 1 && version == "" {
+			buildNumber = tag.Pre[1]
+		}
+
+		tag.Pre = nil
+
+		preVersion, err := semver.NewPRVersion(p.PrereleaseID)
+		if err != nil {
+			log.Errorf("failed to create new pre-release version: %s", err)
+
+			os.Exit(exitcode.ErrDefault)
+		}
+
+		tag.Pre = append(tag.Pre, preVersion)
+
+		buildVersion, err := semver.NewPRVersion(strconv.Itoa(int(buildNumber.VersionNum + 1)))
+		if err != nil {
+			log.Errorf("failed to create new build version: %s", err)
+
+			os.Exit(exitcode.ErrDefault)
+		}
+
+		tag.Pre = append(tag.Pre, buildVersion)
+
+		finalTag = p.Prefix + tag.String()
+	}
+
+	if method == "hotfix" || method == "final" {
+		finalTag = p.Prefix + tag.FinalizeVersion()
+	}
+
+	// Create tag locally and push it.
+	if !p.DryRun {
+		err := ghtag.Create(ghtag.Params{
+			CommitSha:  p.CommitSha,
+			Client:     p.Client,
+			Owner:      p.Owner,
+			Repository: p.Repository,
+			TagMessage: p.TagMessage,
+			Tag:        finalTag,
+		})
+		if err != nil {
+			log.Error(err.Error())
+
+			os.Exit(exitcode.ErrDefault)
+		}
+	}
+
+	fmt.Printf("::set-output name=SEMVER_TAG::%s\n", finalTag)
+
+	os.Exit(exitcode.Success)
+}
+
+// defineBumpStrategy defines the strategy for semver to bump product version.
+func defineBumpStrategy(bump, sourceBranch, destBranch, mainBranchName, developBranchName string) (string, string) {
+	if bump == "auto" {
+		// bugfix into develop branch
+		if branchBugfixPrefixRegex.MatchString(sourceBranch) && destBranch == developBranchName {
+			return "build", "patch"
+		}
```

Comment:
If you make an early return here, you can avoid the indentation level.

```suggestion
	if bump != "auto" {
	    return bump, ""
	}
	
	// bugfix into develop branch
	if branchBugfixPrefixRegex.MatchString(sourceBranch) && destBranch == developBranchName {
		return "build", "patch"
	}
```